### PR TITLE
Gem and testing cleanup

### PR DESCRIPTION
**Pull request by genebean**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 AllCops:
+  TargetRubyVersion: 2.4 # this should match the oldest version in .github/workflows/tests.yaml's matrix
```
Comment (neomilium) [inline .rubocop.yml:2]: :+1:

File: Rakefile [hunk 2]
```diff
@@ -44,11 +45,13 @@ task :syntax do
   end
 end
 
-task :rubocop do
-  require 'rubocop'
-  cli = RuboCop::CLI.new
-  exit_code = cli.run(%w(--display-cop-names --format simple))
-  raise "RuboCop detected offenses" if exit_code != 0
+RuboCop::RakeTask.new(:rubocop) do |task|
```
Comment (neomilium) [inline Rakefile:48]: Its cleaner!

File: onceover.gemspec [hunk 3]
```diff
@@ -36,4 +36,8 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency 'terminal-table', '>= 1.8.0'
   s.add_runtime_dependency 'versionomy', '>= 0.5.0'
 
+  s.add_development_dependency 'cucumber', '~> 4.1'
```
Comment (neomilium) [inline onceover.gemspec:39]: Doing it this way is far cleaner! :+1:

**Review by neomilium — APPROVED** [review]

File: Rakefile (hunk 2 above)
Comment (genebean) [reply Rakefile:48]: This plus the require above also make tasks from Rubocop show up in `bundle exec rake -T`

File: onceover.gemspec (hunk 3 above)
Comment (genebean) [reply onceover.gemspec:39]: single source of truth and all :)

**Review by dylanratcliffe — APPROVED** [review]

**Pull request merged**